packages/gatsby-source-contentful/src/fetch.js

```diff
@@ -39,6 +39,11 @@ module.exports = async ({ syncToken, reporter, pluginConfig }) => {
     let errors
     if (e.code === `ENOTFOUND`) {
       details = `You seem to be offline`
+    } else if (e.code === `SELF_SIGNED_CERT_IN_CHAIN`) {
+      details = `Your self signed certificate for SSL communication is failing. 
+                If you are in development environment you might add NODE_TLS_REJECT_UNAUTHORIZED=0 in your .env file.
```

Comment:
`.env` isn't used by default (some starters might support it, but not all).

We have similar advice in wordpress plugin - https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-source-wordpress/README.md#self-signed-certificates and reading `.env.development` (to make sure this happens only in development scenario) part is covered in https://www.gatsbyjs.org/docs/environment-variables/#server-side-nodejs

Reply:
Hi @pieh in my last commit i used the reporter.panic to show the message and the error object. I've omitted the sentence about .env to be more clear and avoid misunderstanding in case it's not used.